src/game_objects/player.cpp

```diff
@@ -150,3 +160,61 @@ void Player::setup_player_model(std::string &model_name) {
     else if (model_name == "cat")
         transform.set_scale({ 0.004f, 0.004f, 0.004f });
 }
+
+bool Player::s_take_damage() {
+    if (invulnerable)
+        return false;
+
+    // Period of invulerability
+    invulnerable = true;
+
+    // Period of stunned
+    stunned = true;
+
+    std::cerr << "Player is hit: " << id << std::endl;
+
+    // Player should drop gold and lose gold somewhere here
+
+    // End Stunned
+    Timer::get()->do_after(std::chrono::milliseconds(STUN_LENGTH),
+        [&]() {
+        stunned = false;
+    });
+
+    // End Invulernability
+    Timer::get()->do_after(std::chrono::milliseconds(INVULNERABLE_LENGTH),
+        [&]() {
+        invulnerable = false;
+    });
+
+    return true;
+}
+
+void Player::c_take_damage() {
+
+    int count = 0;
+    end_flicker = false;
+
+    // Flicker
+    cancel_flicker = Timer::get()->do_every(
+        std::chrono::milliseconds(100),
+        [&, count]() mutable {
+        if (count % 2)
+            set_filter_alpha(1.0f);
+        else
+            set_filter_alpha(0.2f);
+
+        if (end_flicker)
+            disable_filter();
```

Comment:
Seems like you're doing this twice: once hee, once in the `do_after` on line 214.